### DISCUSSION TITLE
Theme Generator

### DISCUSSION THREAD

**Discussion by Sarenor** [skeletonlabs/skeleton#1141] [Ideas]
I know there's lots of work going into the Theme Generator colorwise (thank you Rhys and Skyphyr!) but we've talked about a couple of things on stream yesterday and I had some thoughts that I wanted to note down somewhere.
To unfinished for an issue though.

Features that were requested and thought about:
1) Locking Colors to not be randomized
2) Being able to edit specific color swatches
3) Saving & Loading themes

Number 1) should probably be easy-ish to implement, but I'm a bit more interested in 2) and 3): Being able to edit specific color swatches is probably a prerequisite for saving & loading of themes since users are of course encouraged to change specific colors and not just the mid point, so we'd have to be able to load & edit & store those again.

As for saving & loading, lazy me would love to just introduce a .json format for it, but then users would need another file in their project just to be able to edit their theme at a later date and we've already got a bunch of .postcss themes out there anyways.   
Especially with v1.0 out of the door, it'll probably be better if users could just load their existing .postcss themes into the editor, get any updated design token values we added integrated into their file so they can tweak it again, save it and savely update the library.

Sorry, bit rambly, but I wanted to kind of collect my thoughts and Chris always tells us that if something is not on Github, it's not permanent ;)

**Comment by endigo9740** [comment]
https://github.com/skeletonlabs/skeleton/issues/1142
